### PR TITLE
Solaris 11.1 support

### PR DESCRIPTION
**Pull request by Phil-Friderici**
based on @juliengk https://github.com/ghoneycutt/puppet-module-pam/pull/113

Fixed the spec tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ This module has been tested to work on the following systems using Puppet v3 wit
  * EL 7
  * Solaris 9
  * Solaris 10
- * Solaris 11
+ * Solaris 11.1
```
Comment (ghoneycutt) [inline README.md]: Is 11.1 different than 11 ?

**Comment by ghoneycutt** [comment]
It appears we are dropping support for 11 while adding support for 11.1. Could you explain?

File: README.md (hunk 1 above)
Comment (Phil-Friderici) [inline README.md]: got this information from @nalyanyam while he tested this code
> using the Solaris 11.1 configuration on 11.0 had real bad consequences – (such as login
> without password, passwd command fail etc).

**Comment by Phil-Friderici** [comment]
got this information from @nalyanyam while he tested this code
> However, there is a difference in pam configuration files between earlier version of Solaris - 11.0
> and Solaris 11.1 Solaris 11.0 has same configuration as earlier OS kernelrealses such as Solaris 10
> – meaning it requires /etc/pam.conf. While beginning form Solaris 11.1, there is an alternative > configuration that uses /etc/pam.d/<service> design

@juliengk could you please explain why Solaris 11.0 now fails when it could be treated like Solaris 10 ?

**Comment by nalyanyam** [comment]
@juliengk, Probably you could update the module such that when the Kernelversion is 11.0, it uses same pam configuration as Solaris 10. Should be similar but i will need to test it again after you have done the changes. 

PAM configuration via the per-service PAM policy files in the /etc/pam.d directory in addition to the pam.conf file, is supported starting from Solaris 11.1

**Comment by juliengk** [comment]
@Phil-Friderici i don't have time that's why!

**Comment by ghoneycutt** [comment]
Closing due to lack of response. Please feel free to re-open.

**Pull request closed without merging**